CMakeLists.txt

```diff
@@ -111,16 +110,24 @@ if(LIBCORO_FEATURE_NETWORKING)
     endif()
 endif()
 
+if(DEFINED EMSCRIPTEN)
+  add_compile_options(-fwasm-exceptions)
+  add_compile_options(-pthread)
+  add_compile_options(-matomics)
+
+  add_link_options(-fwasm-exceptions)
+  add_link_options(-pthread)
+  add_link_options(-sPROXY_TO_PTHREAD)
+  add_link_options(-sALLOW_MEMORY_GROWTH)
+  add_link_options(-sEXIT_RUNTIME)
+endif()
+
 add_library(${PROJECT_NAME} STATIC ${LIBCORO_SOURCE_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX PREFIX "")
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
 target_include_directories(${PROJECT_NAME} PUBLIC include)
-if(MSVC)
-    # Not sure why this is only needed on Windows.
-    target_include_directories(${PROJECT_NAME} PUBLIC vendor/tartanllama/expected/include)
-endif()
+
 if(LIBCORO_FEATURE_PLATFORM)
-    target_link_libraries(${PROJECT_NAME} PUBLIC pthread tl::expected)
```

Comment:
did pthread get added somewhere else consistently? I only see it for EMSCRIPTEN

Reply:
Oops. I'll fix that.